### DISCUSSION TITLE
更新atomci时，如何同步配置到conf中

### DISCUSSION THREAD

**Discussion by sampsonye** [go-atomci/atomci#92] [Ideas]
## 方案一
**不同步** ，在项目开发过程中，新的配置维护到`app.conf.template`文件中，用户可自行查阅

## 方案二 
**文件对比并增量覆盖**，方案大致如下：
1. 维护`app.conf.template`文件，保持配置最新，并跟随项目一同发布
2. 引入开源ini模块(如：[github.com/go-ini/ini](https://github.com/go-ini/ini))，项目初始化时(beego初始化之前)读取`app.conf.template`和`app.conf`
3. 对比两个配置文件中缺少的Section和Key，并写入`app.conf`中
4. 保存最新配置，正常启动项目

## 方案三
**迁移到数据库**，除基础配置（比如httpPort和DB::url）外，其他所有业务配置均迁移到数据库

**Comment by sampsonye** [comment]
@colynn @fanhousanbu @CraiGZero 一起参与讨论下，如何设计更好的“**配置升级**”方案呢

**Reply by fanhousanbu** [reply]
方案一是底线吧？方案二重点是diff ，冲突如何解决？以及废弃的配置如何表达？

**Reply by sampsonye** [reply]
方案二 考虑做简单的diff，为保证业务正确性，可以考虑对比app.conf.template和app.conf时仅考虑template文件中新增的项，并写到app.conf中，原则上不会存在冲突情况

**Reply by fanhousanbu** [reply]
综上所述，方案三获得最高票

**Comment by colynn** [comment]
支持方案一，觉得不是像数据migrate那样很关键的功能，目前配置文件还很有可能发生变化，对于配置文件的灵活性、易用性是功能稳定后再整体考虑是不是更好些，现阶段建议采用方案一
